src/java/org/apache/cassandra/db/tries/InMemoryTrie.java

```diff
@@ -159,8 +198,7 @@ final void putByte(int pos, byte value)
         getChunk(pos).putByte(inChunkPointer(pos), value);
     }
 
-
-    private int allocateBlock() throws SpaceExhaustedException
+    private int allocateNewCell() throws SpaceExhaustedException
```

Comment:
The former is used by the allocator to get a new cell, the latter by the trie to get a cell through the allocator. Because the latter can possibly be recycled, it needs to be cleaned.

Added comments to explain the difference.